UI implementation

obedjamir opened the pull request:
ERROR passing data from form to controller

obedjamir committed:
UI implementation
ERROR passing data from form to controller

obedjamir committed:
Basic UI implementation
home page
login and register page
user account page
update details page

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/CoreModule.java:
```diff
@@ -0,0 +1,13 @@
+
+@SpringBootApplication
+@EnableJpaAuditing
+public class CoreModule {public static void main(String[] args) {
```
This is not required. A single runner in service should be fine

obedjamir replied:
THis has been implemented

PinakiGhosh replied:
This file itself should not be required.

obedjamir replied:
Oh okay i'll remove it in the next commit.

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/RegisterController.java:
```diff
@@ -0,0 +1,53 @@
+@Controller
+@RequestMapping("/register")
+public class RegisterController {
+    @Autowired()
```
() is not required

obedjamir replied:
this has been removed

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/RegisterController.java:
```diff
@@ -0,0 +1,53 @@
+
+    public void authWithHttpServletRequest(HttpServletRequest request, String username, String password) {
+        try {
+            System.out.println(username);
```
You should not be using System.out.println in projects. Use loggers.
Check @Slf4j annotation

obedjamir replied:
this has been implemented

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/RegisterController.java:
```diff
@@ -0,0 +1,53 @@
+        Users response = service.saveUsers(user);
+        if(response==null)
+        {
+            return "email_already_registered";
```
This should be shown as an error message in the screen

obedjamir replied:
I have implemented in such a way that the alert message is shown in the same page.

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/RegisterController.java:
```diff
@@ -0,0 +1,53 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+import static org.hibernate.tool.schema.SchemaToolingLogging.LOGGER;
```
Use @Slf4j. Spring boot comes with logback logger

obedjamir replied:
@Sl4j used to logging

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/TripActivityController.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<Optional<TripActivity>> findTripById(@RequestParam Long id) {
```
You should not be returning optional

obedjamir replied:
This has been changed to return object

obedjamir committed:
core module update
last modified date is updated with every update

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/TripController.java:
```diff
@@ -0,0 +1,56 @@
+import javax.validation.Valid;
+import java.util.Optional;
+@RestController
+@RequestMapping("/trip")
```
Make this as trips.

obedjamir replied:
changed to trips

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/controller/TripController.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<Optional<Trip>> findTripById(@RequestParam Long id) {
```
No optional

obedjamir replied:
this has been updated

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/RegisterManager.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.Optional;
+
+@Service
+public class RegisterManager {
```
The managers should not be basis the functions, but basis the entity.

For all User related task it should be in UserManager. There should not be anything like RegisterManager

obedjamir replied:
register controller and manager removed

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/RegisterManager.java:
```diff
@@ -0,0 +1,38 @@
+        Optional<Users> user = userRepository.findByEmail(data.getEmail());
+        if(user.isEmpty())
+        {
+            data.setCreatedTime(LocalDate.now());
```
We are already using the @Created/Modified annotations. So this is not required

obedjamir replied:
Yes, i was unaware of its functionality. I have removed the manual declarations.

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -20,11 +21,11 @@ public TripActivity saveTripActivity(TripActivity data)
         }
         else{
-            Optional<TripActivity> presentData = exists(data.getTripId());
+            Optional<TripActivity> presentData = exists(data.getTripId().getTripId());
```
Why is this data.getTripId().getTripId() ?

obedjamir replied:
data.getTripId returns an object because it is a foreign key. so i used getTrip() again to get id from object

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -20,11 +21,11 @@ public TripActivity saveTripActivity(TripActivity data)
             {
-                TripActivity newData = updateData(presentData.get(), data);
-                return tripActivityRepository.save(newData);
+                updateData(presentData.get(), data);
```
Ideally a method should not be updating an input parameter. The update data should create a new object, merge the two data and then send it back.

Also is the update method actually updating data or merging 2 objects? Update should ideally save to db as well

obedjamir replied:
here the input parameter is not updated. i update an object of the existing details using the input parameter and update that.

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java:
```diff
@@ -55,25 +56,23 @@ private Optional<TripActivity> exists(Long id)
             presentData.setActivityStatus(tripActivity.getActivityStatus());
         }
-        if(tripActivity.getActivityDescription()!=null)
+        if(!tripActivity.getActivityDescription().isBlank())
```
This will throw NullPointerException if getActivityDescription returns null

obedjamir replied:
all class members are set to nullabale=false. wouldn't that remove any case of it being null?

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/manager/TripManager.java:
```diff
@@ -23,8 +25,8 @@ public Trip saveTrip(Trip data)
             {
-                Trip newData = updateData(presentData.get(),data);
-                return tripRepository.save(newData);
+                updateData(presentData.get(),data);
```
Same as previous

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java:
```diff
@@ -19,7 +19,7 @@
 public class Trip implements Serializable {
     @Id
-    @Column(name="trip_id",nullable = false)
+    @Column(name="trip_id")
```
Why remove nullable?

obedjamir replied:
Oh i was going to use @Generatedvalues there

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java:
```diff
@@ -32,17 +32,17 @@ public class Trip implements Serializable {
     private String tripDescription;
 
     @Column(name="start_date",nullable = false)
-    private Date startDate;
```
Why change from Date?

obedjamir replied:
Date data type was saving time as 00:00:00 so i changed it to LocalDate. i have changed it back to Date

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java:
```diff
@@ -9,7 +9,7 @@
 
 import javax.persistence.*;
 import java.io.Serializable;
-import java.sql.Date;
```
We generally use java.util.Date

obedjamir replied:
yes i have changed it to java.util.Date

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/repository/UsersRepository.java:
```diff
@@ -4,7 +4,9 @@
+import java.util.Optional;
+
 public interface UsersRepository extends JpaRepository<Users,Long> {
     @Query("SELECT user FROM Users user WHERE user.email = ?1")
```
Use the repository naming convention for these queries. No need for native queries. Check  [guide](https://docs.spring.io/spring-data/data-jpa/docs/2.5.x/reference/html/#jpa.query-methods.query-creation)

obedjamir replied:
I have changed user to u. as per what i have seen in the guide.

PinakiGhosh commented on core/src/main/resources/templates/email_already_registered.html:
```diff
@@ -0,0 +1,13 @@
+</head>
+<body>
+<script>
+alert("This email address is already a registered user")
```
This should not be a different page. See how you can show the error message without changing the page

obedjamir replied:
this has been implemented

obedjamir committed:
trip-manager update
all requested changes implemented

obedjamir committed:
trip-manager update
all requested changes implemented

obedjamir committed:
Project structure update
controllers are now in service module
dto layer is implemented using ModelMapper
